code/modules/mapping/mapping_helpers.dm

```diff
@@ -54,6 +56,35 @@
 	name = "lavaland baseturf editor"
 	baseturf = /turf/open/lava/smooth/lava_land_surface
 
+// Does the same thing as baseturf_helper but only the specified kinds of turf (the kind it's placed on or varedited)
+/obj/effect/baseturf_helper/picky
+	var/list/whitelist
+	// Can be mapedited as: a single type, a list of types, or a typecache-like list
+	// The first 2 make a typecache of the given values
+	// The last uses it as is
+
+/obj/effect/baseturf_helper/picky/Initialize()
+	if(!whitelist)
+		whitelist = list(loc.type)
+	else if(!islist(whitelist))
+		whitelist = list(whitelist)
+	else if(whitelist[whitelist[1]]) // Checking if it's a typecache-like list
+		return ..()
+	whitelist = typecacheof(whitelist)
+	return ..()
```

Comment:
. = ..() first in Initialize

Reply:
cant, this needs to be before the initialization